lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -1103,7 +1103,7 @@ def test_static_badlink_rewrite(self):
         result_fragment = module.render(STUDENT_VIEW)
 
         self.assertIn(
-            '/c4x/{org}/{course}/asset/_file.jpg'.format(
+            '/c4x/{org}/{course}/asset/file.jpg'.format(
```

Comment:
So, it was kind of random.  The long and the short of it is that the previous code would turn slashes into underscores so that it didn't break the generated asset key, as far as I could tell.  This looks like the intention was supposed to be that a "bad link" of having a leading forward slash should be sanitized correctly.

The way the new canonicalize method goes and sanitizes incoming paths before doing anything, it implicitly takes care of that.

I might be overlooking something with that assessment/decision, but it didn't seem terrible that the new code could potentially fix a human error automatically, even if unintentionally.

Thoughts?